### ISSUE TITLE
Inference on input image

### ISSUE THREAD

**Issue by MoaazAbdulrahman** [tianyu0207/PEBAL#3]
Thank you for your great work.

Is there a way to run inference on single image and get output mask visualization?

Best regards,

**Comment by yyliu01** [comment]
Thanks for your interest.

You can visualise the **anomaly_score**, which is the predicted anomalous mask.
Btw, we provide the visualisation code in [here](https://github.com/tianyu0207/PEBAL/blob/41d6ccf94daaceeff5e4d9dbfb813314a36f15b2/code/valid.py#L33) for wandb uploading. 

Best regards,

**Comment by yyliu01** [comment]
I'm closing the issue, and feel free to reopen it if your question is not answered.

**Comment by MoaazAbdulrahman** [comment]
Thank you for your response. 
I have tried to save the **anomaly_score** using opencv but all I have got is a black image. I printed the values of the anomaly_score and I got negative values as following:
```
[[-2.9662938 -2.9644353 -2.9604588 ... -2.9755359 -2.9628184 -2.9559977]
 [-2.9675057 -2.965737  -2.9618487 ... -2.9762902 -2.9629307 -2.9557352]
 [-2.9698122 -2.9683406 -2.9649205 ... -2.9781318 -2.9634345 -2.9554574]
 ...
 [-2.9481578 -2.948013  -2.9478688 ... -2.950601  -2.9491034 -2.9483685]
 [-2.9472773 -2.9471524 -2.9470356 ... -2.9499426 -2.94843   -2.9476068]
 [-2.9467642 -2.9466426 -2.9465258 ... -2.9491653 -2.9477067 -2.9468539]]
```

**Comment by yyliu01** [comment]
@[MoaazAbdulrahman](https://github.com/MoaazAbdulrahman),  before you save it, could you please visualise it based on the matplotlib.pyplt to check the correctness (as it provides a built-in normalisation for the input value)?

``` python
import matplotlib.pyplot as plt
plt.imshow(anomaly_score.cpu().numpy())
plt.show()
```

Btw, the anomaly panel is supposed to be all negative value, which is calculated via energy score (e.g., -torch.logsumexp()).

**Comment by MoaazAbdulrahman** [comment]
Thanks it works now.
